pkg/graveler/staging/manager.go

```diff
@@ -10,15 +10,24 @@ import (
 )
 
 type Manager struct {
-	store kv.StoreMessage
-	log   logging.Logger
+	store  kv.StoreMessage
+	log    logging.Logger
+	wakeup chan bool
+
+	// notifiers are being called with every successful cleanup cycle
+	notifiers []chan bool
 }
 
-func NewManager(store kv.StoreMessage) *Manager {
-	return &Manager{
-		store: store,
-		log:   logging.Default().WithField("service_name", "staging_manager"),
+func NewManager(ctx context.Context, store kv.StoreMessage, notifiers ...chan bool) *Manager {
+	const wakupChanCapacity = 100
+	m := &Manager{
+		store:     store,
+		log:       logging.Default().WithField("service_name", "staging_manager"),
+		wakeup:    make(chan bool, wakupChanCapacity),
```

Comment:
I think sync.Cond will fit this one better. we like to signal clean and not trigger it based on the number of times we called drop.

Reply:
synced f2f, doesn't fit since we don't want to split the shutdown and cleanup flows (sync.Condition isn't waitable in a select statement)